From b24bbe7a636e6b240ea47ea14ae40c4ec75addda Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Mon, 31 Dec 2012 12:50:16 -0600
Subject: [PATCH 395/486] ENGR00238582: Compile Jellybean 4.2 for iMX53 SMD

Fix the issue when compiling with GCC 4.6 -
"error: call to char* __builtin___strncat_chk(char*, char const*,
unsigned int, unsigned int) might overflow destination buffer."

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 mx5x/libcamera/CameraHal.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mx5x/libcamera/CameraHal.cpp b/mx5x/libcamera/CameraHal.cpp
index 111c4ba..1be1c8b 100755
--- a/mx5x/libcamera/CameraHal.cpp
+++ b/mx5x/libcamera/CameraHal.cpp
@@ -454,7 +454,7 @@ using namespace android;
     CAMERA_HAL_RET CameraHal :: InitPictureExifParam(CameraParameters *pParam)
     {
         CAMERA_LOG_FUNC;
-        char tmpBuffer[CAMER_PARAM_BUFFER_SIZE];
+        char tmpBuffer[CAMER_PARAM_BUFFER_SIZE+1];
 
         /*hard code here*/
         pParam->set(CameraParameters::KEY_FOCUS_DISTANCES, "24.0,50.0,2147483648.0");
-- 
1.8.0

